services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/service_tracker/_api.py

```diff
@@ -0,0 +1,249 @@
+from models_library.users import UserID
+from servicelib.deferred_tasks import TaskUID
+
+from ._models import SchedulerServiceState, TrackedServiceModel, UserRequestedState
+from ._setup import get_tracker
+from ._tracker import Tracker
+
+_logger = logging.getLogger(__name__)
+
+
+_LOW_RATE_POLL_INTERVAL: Final[timedelta] = timedelta(seconds=1)
+NORMAL_RATE_POLL_INTERVAL: Final[timedelta] = timedelta(seconds=5)
+_MAX_PERIOD_WITHOUT_SERVICE_STATUS_UPDATES: Final[timedelta] = timedelta(seconds=60)
+
+
+async def set_request_as_running(
+    app: FastAPI,
+    dynamic_service_start: DynamicServiceStart,
+) -> None:
+    """Stores the intention fo the user: ``start`` requested"""
+    tracker: Tracker = get_tracker(app)
+
+    node_id: NodeID = dynamic_service_start.node_uuid
+
+    model: TrackedServiceModel | None = await tracker.load(node_id)
+    if model is not None:
+        model.dynamic_service_start = dynamic_service_start
+        model.requested_state = UserRequestedState.RUNNING
+        model.project_id = dynamic_service_start.project_id
+        model.user_id = dynamic_service_start.user_id
+
+    await tracker.save(
+        node_id,
+        TrackedServiceModel(
+            dynamic_service_start=dynamic_service_start,
+            requested_state=UserRequestedState.RUNNING,
+            project_id=dynamic_service_start.project_id,
+            user_id=dynamic_service_start.user_id,
+        ),
+    )
+
+
+async def set_request_as_stopped(
+    app: FastAPI, dynamic_service_stop: DynamicServiceStop
+) -> None:
+    """Stores the intention of the user: ``stop`` requested"""
+    tracker: Tracker = get_tracker(app)
+    model: TrackedServiceModel | None = await tracker.load(dynamic_service_stop.node_id)
+
+    if model is None:
+        model = TrackedServiceModel(
+            dynamic_service_start=None,
+            user_id=dynamic_service_stop.user_id,
+            project_id=dynamic_service_stop.project_id,
+            requested_state=UserRequestedState.STOPPED,
+        )
+
+    model.requested_state = UserRequestedState.STOPPED
+    await tracker.save(dynamic_service_stop.node_id, model)
+
+
+def __get_state_str(status: NodeGet | DynamicServiceGet | NodeGetIdle) -> str:
+    # Attributes where to find the state
+    # NodeGet -> service_state
+    # DynamicServiceGet -> state
+    # NodeGetIdle -> service_state
+    state_key = "state" if isinstance(status, DynamicServiceGet) else "service_state"
+
+    state: ServiceState | str = getattr(status, state_key)
+    result: str = state.value if isinstance(state, ServiceState) else state
+    return result
+
+
+def _get_poll_interval(status: NodeGet | DynamicServiceGet | NodeGetIdle) -> timedelta:
+    if __get_state_str(status) != "running":
+        return _LOW_RATE_POLL_INTERVAL
+
+    return NORMAL_RATE_POLL_INTERVAL
+
+
+def _get_current_state(
+    requested_sate: UserRequestedState,
+    status: NodeGet | DynamicServiceGet | NodeGetIdle,
```

Comment:
would it not be simpler to handle the different status once and for all and have then functions that handle just one state instead of creating if cases everywhere?